core/modules/cart/store/actions.ts

```diff
@@ -492,12 +492,12 @@ const actions: ActionTree<CartState, RootState> = {
           headers: { 'Content-Type': 'application/json' },
           mode: 'cors'
         },
-        silent: false
+        silent: true
       })
       if (task.result === true) {
         dispatch('syncTotals', { forceServerSync: true })
       }
-      return task.result
+      return task
```

Comment:
I'd stay here with returning result, but wrapping everything with try-catch block and return false when code is incorrect.